src/cache_cf.cc

```diff
@@ -1115,18 +1115,19 @@ parseTimeUnit(const char *unitName, std::chrono::nanoseconds &ns)
 }
 
 static std::chrono::nanoseconds
-CheckTimeValue(const double value, const std::chrono::nanoseconds &unit)
+ConvertToNanoSeconds(const double value, const std::chrono::nanoseconds &unit)
 {
-    if (value < 0)
+    if (value < 0.0)
         throw TexcHere("time must have a positive value");
 
-    const auto maxNanoseconds = std::chrono::nanoseconds::max().count();
-    if (value > maxNanoseconds/static_cast<double>(unit.count())) {
-        const auto maxYears = maxNanoseconds/(HoursPerYear*3600*1000000000);
+    const auto result = static_cast<std::chrono::nanoseconds::rep>(value * unit.count());
```

Comment:
We need to check for std::chrono::nanoseconds::rep overflow _before_ we forcefully cast to that type. Checking _after_ the forced conversion is pointless -- the overflow, if any, would be removed by that conversion.

Reply:
ok. Found a better way, I think. Please see updated branch